corelibrary/src/main/java/com/nullgr/corelibrary/date/DateFormatExtensions.kt

```diff
@@ -17,59 +21,128 @@ object SimpleDateFormatterCache {
     operator fun set(name: String, typeface: SimpleDateFormat) {
         cache.put(name, typeface)
     }
+
+    fun getOrCreateFormatter(dateFormat: String, timeZone: TimeZone? = null, locale: Locale? = Locale.getDefault()): SimpleDateFormat {
+        val key = "$dateFormat${locale?.displayName ?: ""}${timeZone?.displayName ?: ""}"
+        var format = this[key]
+        if (format == null) {
```

Comment:
`Locale.getDefault()` passed as default parameter maybe we can simplify this code?